src/storage/persistence/AccountSecureStorage.js

```diff
@@ -28,8 +29,38 @@ export class AccountSecureStorage extends BaseSecureStorage {
     }
 
     /**
-     * Get all the accounts
-     * @returns {Promise<AccountModel[]>}
+     * Updates account
+     * @param newAccount
+     * @returns {Promise<*>}
+     */
+    static async updateAccount(newAccount: AccountModel): Promise<any> {
+        const allAccounts = await this.getAllAccounts();
+        const edited = allAccounts.map(account => {
+            if (account.id === newAccount.id) {
+                return newAccount;
+            } else {
+                return account;
+            }
+        });
+        return this.saveAccounts(edited);
+    }
+
+    /**
+     * Removes an account by its id and network
+     * @param id
+     * @param network
+     * @returns {Promise<*>}
+     */
+    static async removeAccount(id: string, network: AppNetworkType): Promise<AccountModel[]> {
+        const allAccounts = await this.getAllAccounts();
+        const filteredAccounts = allAccounts.filter(account => !(account.id === id && account.network === network));
```

Comment:
Does the `updateAccount` also need filter by networkType? Or the ids are unique for all networks

Reply:
No they are not unique now